Java/core/src/main/java/com/amazon/randomcutforest/anomalydetection/AbstractAttributionVisitor.java

```diff
@@ -121,9 +121,9 @@ public void accept(Node node, int depthOfNode) {
             // use the sibling bounding box to represent counterfactual "what if point & the
             // candidate near neighbor
             // had not been inserted in the tree"
-            Node sibling = Node.isLeftOf(pointToScore, node) ? node.getRightChild() : node.getLeftChild();
 
-            shadowBox = shadowBox == null ? sibling.getBoundingBox() : shadowBox.getMergedBox(sibling.getBoundingBox());
+            shadowBox = shadowBox == null ? Node.getSiblingBoundingBox(pointToScore, node)
```

Comment:
discussion. Should this method be non-static given the node instance?
it can be node.sib(point) or even node.sib(point, shadow)

Reply:
The answer depends on the goal :) For the current context, these three are equivalent. But for CompactRCF, there is no actual node -- the node in this function call would become an instance of INodeView (other suggestions welcome!). The  CompactRCFTree would create a class that provides this interface and spoofs the nodes without creating transient data  structures (which would mean more frequent GC) -- that is the current scenario which we are trying to avoid.

Reply:
But we can also make it node.getSiblingBoundingBox(pointToScore)  which would become nodeView.getSiblingBoundingBox(pointToScore). If we do that would we not increase the size of the Node class for the default context? Admittedly, that may not matter ... but still

Reply:
i am not clear how making it a static method helps compact rcf

Reply:
Static/Non-Static will not matter to CompactRCF. But non-static may increase the space required by the Node class and be detrimental to the current non-compact algorithm. I am happy to make it non-static. @jotok?

Reply:
Actually making it non-static helps parts 2 and 3 of the PR for CompactRCF. Going with node.getSiblingBox() ...